### PR TITLE
Add test for skipping empty patches during import --flatten

### PR DESCRIPTION
**Pull request by gyoshev**
Currently, `lerna import --flatten` fails to import repositories with empty commits. This PR fixes this by detecting the case and skipping any empty commits.

## Description

This patch skips empty commits when `lerna import --flatten` is used. This is done by detecting empty patches and using `git am --skip`.

## Motivation and Context

Using `lerna import --flatten` on repos with empty commits yields a cryptic error, because `git am` does not support importing of empty commits:

```
lerna ERR! execute callback with error
(node:24090) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot read property 'split' of undefined 
```

## How Has This Been Tested?

I added a unit test, and used the new import on a repository that previously failed with the above error.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/commands/ImportCommand.js [hunk 1]
```diff
@@ -171,6 +171,11 @@ export default class ImportCommand extends Command {
       // Fall back to three-way merge, which can help with duplicate commits
       // due to merge history.
       ChildProcessUtilities.exec("git", ["am", "-3"], this.execOpts, (err) => {
+        if (err && /^patch is empty/i.test(err.stdout)) {
```
Comment (evocateur) [inline src/commands/ImportCommand.js]: This was already fixed in https://github.com/lerna/lerna/pull/1059/files#diff-84077e6291e8b6b2a596aff7f258b0dd, but the PR view doesn't seem to incorporate those changes.

I like your unit test, but you need to rebase and remove this block before I can merge it.

File: test/ImportCommand.js [hunk 2]
```diff
@@ -109,6 +109,22 @@ describe("ImportCommand", () => {
       expect(await pathExists(newFilePath)).toBe(true);
     });
 
+    it("skips empty patches with --flatten", async () => {
```
Comment (evocateur) [inline test/ImportCommand.js]: Does it _require_ `--flatten` to skip the empty commits? The code (here or in #1059) doesn't seem to imply this.

Comment (gyoshev) [reply test/ImportCommand.js]: If I understand this correctly, the problem occurs only with `--flatten`, because `createPatchForCommit` uses `git log`, instead of `git format-patch`. These have different behavior for empty commits:

```
(~) git format-patch -1 f3121c --stdout                                             
(~) git log --reverse --first-parent -p -m --pretty=email --stat --binary -1 f3121c 
From f3121c... Mon Sep 17 00:00:00 2001
From: Alex Gyoshev <___@___.__>
Date: Thu, 12 Oct 2017 14:23:10 +0300
Subject: ...
(~) 
```

**Comment by gyoshev** [comment]
I've rebased the branch and left only the test case. The test probably should have been along with the other PR, but at least now there is some protection against regressions.

File: test/ImportCommand.js (hunk 2 above)
Comment (evocateur) [reply test/ImportCommand.js]: Oh, interesting! We should probably address this disparity in the future.

**Comment by evocateur** [comment]
Thanks @gyoshev!
